pkg/util/stringencoding/string_encoding.go

```diff
@@ -118,17 +125,21 @@ const uppercaseHex = `0123456789ABCDEF`
 // writeMultibyteRuneAsHex is equivalent to either
 // fmt.FPrintf(`\u%04X`) or fmt.FPrintf(`\U%08X`).
 // We can't quite just use strconv since we need uppercase hex.
-func writeMultibyteRuneAsHex(buf *bytes.Buffer, r rune, ln int) {
+func writeMultibyteRuneAsHex(buf Buffer, r rune, ln int) {
+	const mask = 0x0f
 	if ln == 2 {
-		buf.WriteString(`\u0000`)
+		buf.WriteString(`\u`)
 	} else {
-		buf.WriteString(`\U00000000`)
+		buf.WriteString(`\U`)
+		buf.WriteByte(uppercaseHex[r>>28&mask])
+		buf.WriteByte(uppercaseHex[r>>24&mask])
+		buf.WriteByte(uppercaseHex[r>>20&mask])
+		buf.WriteByte(uppercaseHex[r>>16&mask])
 	}
-	for i := 1; r > 0; r >>= 4 {
-		buf.Bytes()[buf.Len()-i] = uppercaseHex[r&0x0f]
```

Comment:
`Bytes()` only exists in `bytes.Buffer` so this replaces it with a equivalent code from `io.ByteWriter`.